File: Medium/1424. Diagonal Traverse II/1424. Diagonal Traverse II.py
from typing import List

class Solution:
    def findDiagonalOrder(self, nums: List[List[int]]) -> List[int]:
        n = len(nums)
        temp = []

        for i in range(n):
            m = len(nums[i])
            for j in range(m):
                ti = i + j

                if len(temp) <= ti:
                    temp.append([])
                temp[ti].insert(0, nums[i][j])

        result = []
        for row in temp:
            result.extend(row)

        return result

File: Medium/1424. Diagonal Traverse II/1424. Diagonal Traverse II.rs
use std::collections::*;

impl Solution {
    pub fn find_diagonal_order(nums: Vec<Vec<i32>>) -> Vec<i32> {
        let n = nums.len();
        let mut temp = vec![];

        for i in 0..n {
            let m = nums[i].len();
            for j in 0..m {
                let ti = i + j;

                if temp.len() <= ti {
                    temp.push(VecDeque::new());
                }
                temp[ti].push_front(nums[i][j]);
            }
        }

        let mut result = vec![];
        for que in &temp {
            for &v in que {
                result.push(v);
            }
        }
        result
    }
}
